### PR TITLE
IGAPP-642: Refactor feedback module

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: web-ts/src/components/FeedbackBoxContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  closeFeedbackModal: () => void
+  sendingStatus: SendingStatusType
+  onSubmit: (sendingStatus: SendingStatusType) => void
+  t: TFunction
+}
+
+/**
+ * Renders a FeedbackBox with all possible feedback options the User can select
+ */
+export const FeedbackBoxContainer = ({
```
Comment (Taggotty) [inline web-ts/src/components/FeedbackBoxContainer.tsx:34]: Also converted to functional component to use `useParams` and `useLocation`

File: web-ts/src/components/FeedbackBoxContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    try {
+      await createFeedbackEndpoint(cmsApiBaseUrl).request(feedbackData)
+      onSubmit('SUCCESS')
+    } catch (e) {
+      console.error(e)
+      onSubmit('ERROR')
+    }
+  }, [location, path, alias, isPositiveRatingSelected, comment, contactMail, city, language, onSubmit])
```
Comment (Taggotty) [inline web-ts/src/components/FeedbackBoxContainer.tsx:91]: removed most of the function calls here as I didn't think they don't really add sth in functional components

File: web-ts/src/components/__tests__/FeedbackBoxContainer.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    ...jest.requireActual('api-client'),
+    createFeedbackEndpoint: (baseUrl: string) => ({
+      request: mockRequest
+    })
+  }
+})
+
+// TODO reenable after route setup is finished
+// eslint-disable-next-line jest/no-disabled-tests
+describe.skip('FeedbackBoxContainer', () => {
```
Comment (Taggotty) [inline web-ts/src/components/__tests__/FeedbackBoxContainer.spec.tsx:30]: It makes no sense to test this component until the Routing to Feedback is not yet setup correctly (especially as we previously relied on the redux route payload)

File: web-ts/src/@types/FeedbackRatingType.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export type FeedbackRatingType = 'up' | 'down'
```
Comment (maxammann) [inline web-ts/src/@types/FeedbackRatingType.ts:1]: Why is this in the @types directory and not in the file where the FeedbackRatingType is used?

I understand the @types directory more to be external types, or types for untyped modules.

File: web-ts/src/components/__tests__/ModalHeader.spec.tsx [hunk 5]
```diff
@@ -1,11 +1,8 @@
-// @flow
-
 import React from 'react'
 import { fireEvent, render } from '@testing-library/react'
-
 import { ModalHeader } from '../ModalHeader'
 import { ThemeProvider } from 'styled-components'
-import lightTheme from '../../../theme/constants/theme'
+import { lightTheme } from 'build-configs/integreat/theme'
```
[the remark this answers is not in the document]
Comment (maxammann) [reply web-ts/src/components/__tests__/ModalHeader.spec.tsx:5]: Jup, this import is gone now

**Review by maxammann — APPROVED** [review]

**Review by klinzo — APPROVED** [review]

File: .eslintrc.js [hunk 6]
```diff
@@ -76,6 +76,7 @@ module.exports = {
     {
       files: ['*.spec.{ts,tsx}'],
       rules: {
+        '@typescript-eslint/no-empty-function': 'off',
```
Comment (Taggotty) [inline .eslintrc.js:79]: Imo this rule makes no sense in tests. But feel free to proof me wrong :)

File: web-ts/src/@types/FeedbackRatingType.ts (hunk 4 above)
Comment (Taggotty) [reply web-ts/src/@types/FeedbackRatingType.ts]: Moved it to FeedbackToolbarItem. Originally I thought the rating type will be used at more places, so I moved it to `@types`.

**Pull request merged**